transfers/amclient.py

```diff
@@ -336,6 +355,80 @@ def download_package(self, uuid):
         else:
             LOGGER.warning('Unable to download package %s', uuid)
 
+    def get_pipelines(self):
+        """GET Archivematica Pipelines (dashboard instances from the storage
+        service.
+        """
+        return utils._call_url_json('{0}/api/v2/pipeline/'.format(self.ss_url),
+                                    params=None,
+                                    headers=self._ss_auth_headers())
+
+    def get_transfer_status(self, transfer_uuid):
+        """GET transfer status if there is a transfer in progress in the
+        Archivematica pipeline. If there aren't an transfers then the
+        response from the server will look as follows:
+
+       ``{"message": "Cannot fetch unitTransfer with UUID
+          ebc8a35c-6742-4264-bc30-22e263966d69", "type": "transfer",
+          "error": true}``
+
+        The response suggesting non-existence is an error, which the caller
+        will have to handle appropriately.
+        """
+        return utils._call_url_json(
+            '{0}/api/transfer/status/{1}/'.format(self.am_url, transfer_uuid),
+            headers=self._am_auth_headers(),
+            params=None)
+
+    def get_processing_config(self, processing_name):
+        """GET a processing configuration file from an Archivematica instance.
+        if the request is succesful an application/xml response is returned
+        to the caller. If the request is unsuccessful then an error code is
+        returned which needs to be handled via error_lookup.
+        """
+        return utils._call_url_json(
+            '{0}/api/processing-configuration/{1}'
+            .format(self.am_url,
+                    processing_name),
+            params=None,
+            headers=self._am_auth_headers())
+
+    def approve_transfer(self, directory, transfer_type=TT_STANDARD):
+        """Approve a transfer in the Archivematica Pipeline. The transfer_type
+        informs Archivematica how to continue processing. Options are:
+          * standard
+          * unzipped bag
+          * zipped bag
+          * dspace
+        Directory is the location where the transfer is to be picked up
+        from. The directory can be found via the get_transfer_status API
+        call.
+        """
+        url = '{0}/api/transfer/approve/'.format(self.am_url)
+        params = {"type": transfer_type, "directory": directory}
+        return utils._call_url_json(url,
+                                    headers=self._am_auth_headers(),
+                                    params=params,
+                                    method="POST")
+
+    def reingest_aip(self, pipeline_uuid, aip_uuid, reingest_type="full",
```

Comment:
Good knowledge! I hadn't seen this PR. The main function Holly added that isn't in AMClient yet is https://github.com/artefactual/automation-tools/blob/31a186887fc8886446604e7791710a2bbd21de7f/reingest/reingest.py#L140 `metadata()` which adds metadata to a SIP. Holly didn't write any tests for this, so I think for now we should close this PR and then consider how we might use it if we get a decent piece of work to develop and test against rather than trying to put it in now.